dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerMainIntegrationTest.java

```diff
@@ -64,8 +67,12 @@ public void testJarFiles()
 
     String jarArgument = googleCloudCore + "," + googleCloudFirestore + "," + guava;
 
-    // This should not raise Exception
-    LinkageCheckerMain.main(new String[] {"-j", jarArgument});
+    try {
+      LinkageCheckerMain.main(new String[] {"-j", jarArgument});
+      fail("LinkageCheckerMain should throw LinkageCheckResultException upon errors");
+    } catch (LinkageCheckResultException expected) {
```

Comment:
With `System.exit`, we would not be able to test LinkageCheckerMain's failure status. Therefore this PR proposes an exception to tell the failure to the caller of the tool (Gradle in mind).

Reply:
You could extract everything from main except System.exit() into a new method and test that. Or you could add an extra "don't exit" argument for tests, or some such. But this should work too.